### PR TITLE
Ownership change, author additions, remove memoization

### PR DESCRIPTION
**Pull request by rpearce**
* changes ownership from Robert Pearce to Articulate Global, Inc., most importantly in `LICENSE` and in `package.json`
* adds `mgreystone` and `jsilve` to `AUTHORS` and "all-contributors"
* bumps `devDependencies`
* removes memoization
* moves `jest` config to `jest.config.js` file
* moves component to `index.js` file to get rid of fluff of an extra file

After merge:
- [x] update changelog to reflect 👆 
- [x] bump to new major version because of transfer
- [x] publish to NPM
- [x] cut new release on GitHub

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,11 +5,11 @@
...
   },
-  "homepage": "https://github.com/rpearce/react-with-responsive",
-  "bugs": "https://github.com/rpearce/react-with-responsive/issues",
-  "author": "Robert Pearce <me@robertwpearce.com>",
+  "homepage": "https://github.com/articulate/react-with-responsive",
+  "bugs": "https://github.com/articulate/react-with-responsive/issues",
+  "author": "Articulate Global, Inc. <engineering@articulate.com>",
   "contributors": [
     "Robert Pearce <me@robertwpearce.com> (https://robertwpearce.com)"
   ],
```
Comment (mgreystone) [inline package.json:18]: ☝️ Nitpick.

File: source/index.js [hunk 2]
```diff
@@ -1 +1,46 @@
-export { default } from './withResponsive'
+import React, { PureComponent } from 'react'
+
+const withResponsive = (
+  query,
+  identifier,
+  displayName='Responsive'
+) => Comp => {
+  class Responsive extends PureComponent {
```
Comment (mgreystone) [inline source/index.js]: This is out of scope... but `PureComponent` has a habit of eating legacy context. Usually not a good fit for a HOC.

**Review by mgreystone — APPROVED** [review]
👍 All looks good to me.

File: source/index.js (hunk 2 above)
Comment (rpearce) [reply source/index.js]: Oh that's really interesting! Have you seen any writeups or github issues regarding this? Or is this from experience?

File: package.json (hunk 1 above)
Comment (rpearce) [reply package.json:18]: You are totally correct. I missed this one

**Review by jsilve — APPROVED** [review]

File: source/index.js (hunk 2 above)
Comment (mgreystone) [reply source/index.js]: https://reactjs.org/docs/legacy-context.html#updating-context
